core/node/consensus/src/en.rs

```diff
@@ -174,6 +162,62 @@ impl EN {
         }
     }
 
+    /// Monitors the `AttestationStatus` on the main node,
+    /// and updates the attestation config accordingly.
+    async fn run_attestation_updater(
+        &self,
+        ctx: &ctx::Ctx,
+        genesis: validator::Genesis,
+        attestation: Arc<attestation::Controller>,
+    ) -> ctx::Result<()> {
+        const POLL_INTERVAL: time::Duration = time::Duration::seconds(5);
+        let Some(committee) = &genesis.attesters else {
+            return Ok(());
+        };
+        let committee = Arc::new(committee.clone());
+        let mut next = attester::BatchNumber(0);
+        loop {
+            let status = loop {
+                match self.fetch_attestation_status(ctx).await {
+                    Err(err) => tracing::warn!("{err:#}"),
+                    Ok(status) => {
+                        if status.genesis != genesis.hash() {
+                            return Err(anyhow::format_err!("genesis mismatch").into());
+                        }
+                        if status.next_batch_to_attest >= next {
+                            break status;
+                        }
+                    }
+                }
+                ctx.sleep(POLL_INTERVAL).await?;
+            };
+            tracing::info!(
+                "waiting for hash of batch {:?}",
+                status.next_batch_to_attest
+            );
+            let hash = self
+                .pool
+                .wait_for_batch_hash(ctx, status.next_batch_to_attest)
+                .await?;
+            tracing::info!(
+                "attesting batch {:?} with hash {hash:?}",
+                status.next_batch_to_attest
+            );
+            attestation
+                .start_attestation(Arc::new(attestation::Info {
+                    batch_to_attest: attester::Batch {
+                        genesis: status.genesis,
+                        hash,
+                        number: status.next_batch_to_attest,
+                    },
+                    committee: committee.clone(),
+                }))
+                .await
+                .context("start_attestation()")?;
+            next = status.next_batch_to_attest.next();
```

Comment:
I can't see the EN saving the certificate any more. Is that deliberate?

Reply:
yes

Reply:
Okay, might want to mention it in the PR description. @RomanBrodetski so from now you can only look for certificates on the main node.

Reply:
added to description